### PR TITLE
chore: cleaner handling of higher order function type inference (styled and forwardRef)

### PR DESCRIPTION
**Pull request by tyleroooo**
I created two utilities for turning react higher order functions into normal higher order functions so the typescript compiler can do higher order type inference correctly.

### CODE REVIEW

**Review by yogurtandjam — APPROVED** [review]

File: src/pages/trade/HorizontalPanel.tsx [hunk 1]
```diff
@@ -308,13 +309,16 @@ export const HorizontalPanel = ({ isOpen = true, setIsOpen }: ElementProps) => {
 const $AssetIcon = styled(AssetIcon)`
   font-size: 1.5em;
 `;
+const collapsibleTabsType = getSimpleStyledOutputType(CollapsibleTabs, {} as {});
```
Comment (jaredvu) [inline src/pages/trade/HorizontalPanel.tsx:312]: `{} as {}` is it possible to make this optional w/ a default in `getSimpleStyledOutputTypes`

File: src/pages/trade/HorizontalPanel.tsx [hunk 2]
```diff
@@ -308,13 +309,16 @@ export const HorizontalPanel = ({ isOpen = true, setIsOpen }: ElementProps) => {
...
+
 const $CollapsibleTabs = styled(CollapsibleTabs)`
   --tableHeader-backgroundColor: var(--color-layer-3);
 
   header {
     background-color: var(--color-layer-2);
   }
-` as typeof CollapsibleTabs<InfoSection>;
+` as typeof collapsibleTabsType;
```
Comment (jaredvu) [inline src/pages/trade/HorizontalPanel.tsx:320]: Do we completely lose the `<InfoSection>` generic?

**Comment by jaredvu** [comment]
Instead of worrying about when we should be casting the return of `styled`, is there a way that we could create some sort of abstraction that handles type inference to use in place of `styled from styled-components`?

File: src/pages/trade/HorizontalPanel.tsx (hunk 2 above)
Comment (tyleroooo) [reply src/pages/trade/HorizontalPanel.tsx:320]: Nope, it gets to be inferred at usage-time, as all generics would prefer.

**Comment by tyleroooo** [comment]
The problem is that sometimes this thing doesn't work either because we're losing all the types styled adds like the string (for nesting) and some other react internal stuff. It's fine here but might break in other places. Given that it's rare to need these generics it feels easiest to special case these instead of pushing the special casing somewhere else.

**Review by jaredvu — APPROVED** [review]

File: src/pages/trade/HorizontalPanel.tsx (hunk 1 above)
Comment (tyleroooo) [reply src/pages/trade/HorizontalPanel.tsx]: Oh yeah I like that. Added!

**Pull request merged**